### ISSUE TITLE
VNC does not work in ver 2.0

### ISSUE THREAD

**Issue by stentmartin** [meefik/linuxdeploy#779]
I have installed both Debian and Kali Linux on my phone on both sdcard0 and sdcard1 (NTFS).
It worked on the old version but not in ver 2.0.
Can not connect to port!? I can not find away around it. 

Please advise.

stentmartin

**Comment by Somebodyhura** [comment]
1_have VNC packages installed completely? 
2_are you connecting to 127.0.0.1 or something else?
3_clear the "port" field in your vnc client or use default port.
4_do you have a firewall installed in your phone?(e.x AFwall+)

**Comment by stentmartin** [comment]
I am using my local IP and using standard ports. No firewall but Arduino
6.0 gives me some security problems with SELinux.

The installation works fine and vnc is started.

But still I can not connect. Tried with three different viewers, same
problem. Can not connect to "Port".

**Comment by Somebodyhura** [comment]
By "Arduino"  you mean Android?

Well I gotta say I'm on Android 6 too.
Installed Kali Linux as partition and I'm using the vnc viewer "com.realvnc.viewer.android"  and it's all good.

I don't think SELinux may be a problem for this issue. (you can use this app for that issue :
https://forum.xda-developers.com/showthread.php?t=2524485)

Can you copy paste here the log and status?

**Comment by stentmartin** [comment]
This is my error messages:

VNC connection faield!

Faild to connect to /192/168/0/198 port 5900:

ECONNREFUSED
(connection refused)

**Comment by Somebodyhura** [comment]
Probably you modem router is doing its job.

Try to connect to this next time:


127.0.0.1:5900

**Comment by stentmartin** [comment]
Hi, I have excluded all other fault and found the problem in the Linux deploy distribution.
It works on my phone with some distributions and settings.

I have given the image file 20GB on my external SD Card (128GB NTFS, use ParagonNTFS, Total Commander and X-plore manager) and I am using kalitop10_arm. 

Have to set SELinux in permissive mode in ParagonNTFS Settings.

I will try to upgrade to GNOME GUI since GNOME is decided to be the Linux future standard (sudo apt-get install xorg, sudo apt-get install task-gnome-desktop).

Best
stentmartin



On May 5, 2017, at 4:30 PM, Somebodyhura <notifications@github.com> wrote:

Probably you modem router is doing its job.

Try to connect to this next time:

127.0.0.1:5900

**Comment by devendra2008** [comment]
I have problem to connect vnc

**Comment by Somebodyhura** [comment]
@stentmartin ok so it was the distribution bug.

Well I never touched SELinux option, yet it's all working for me.

But I recommend you NOT to install GNOME... That's too heavy.
You may get lags and performance decrease.
I'm sure you will get.

That's why there is no GNOME option in linux deploy, it's resource hungry and wasn't worth.

Anyway if you want to try it go ahead, that was just an advice.
:)

**Comment by stentmartin** [comment]
I will test everything even GNOME because I am curios and I am a developer of future medical apps

I have no problem on my internal memory with 4GB container with the Kalitop10_arm with Mate GUI, I have tested it several times and deleted the .img. It works perfect.

But exact the same distribution an my external sc card does not work. I have tested with different sad cards with different installation result. My 128GB Samsung HC EVO can not even install the files correct, many missing files and errors. My 32GB Kingston SD Card installs everything correct without no errors as I can see, but still VNC can not see the port!?
I even have a 512GB SD Card I will test when everything is working.

It is problems everywhere with bad written software and problematic devices.
I have spent weeks to find all the problems and downloaded all recommended + apps from google play. Many of the apps doesn't work correctly either.

You need a lot f time to find out the right setup and software packages. Anyway the only thing that doesn't work right now is Linux on my external SD Card.

I will keep you posted. By the way my Xiaomi Redmi 4 Pro is a fantastic computer/phone.
I give it 5 stars. Only thing missing now is exFAT.

Best

stentmartin

On May 8, 2017, at 3:48 PM, Somebodyhura <notifications@github.com> wrote:

@stentmartin ok so it was the distribution bug.

Well I never touched SELinux option, yet it's all working for me.

But I recommend you NOT to install GNOME... That's too heavy.
You may get lags and performance decrease.
I'm sure you will get.

That's why there is no GNOME option in linux deploy, it's resource hungry and wasn't worth.

Anyway if you want to try it go ahead, that was just an advice.
:)

**Comment by Somebodyhura** [comment]
@stentmartin 

Have you tried my installation guide?
https://github.com/meefik/linuxdeploy/wiki/Installing-Kali-Linux-Guide

That just resolves many issues of not installing.

Read it completely and try it on one of the failed SD card.

I'm pretty sure that will work, somehow it's guaranteed ;D

**Comment by stentmartin** [comment]
Ok, I will try it later tonight. I have just installed the 500GB external sad card.
I gave 20GB to Kali Linux. It toke 42 minutes to install and everything was ok with no errors.
It also configured and installed its internal VNC server correctly.

But the VNC client can still not find and connect to the port!?

So, I will try your installation guide later and let you know what happened. 

Thank you.

stentmartin


On May 8, 2017, at 6:21 PM, Somebodyhura <notifications@github.com> wrote:

@stentmartin

Have you tried my installation guide?
https://github.com/meefik/linuxdeploy/wiki/Installing-Kali-Linux-Guide

That just resolves many issues of not installing.

Read it completely and try it on one of the failed SD card.

I'm pretty sure that will work, somehow it's guaranteed ;D

**Comment by Somebodyhura** [comment]
By the way, you said your developing a medical app...what type? What is it?

And I'm just too curious...why an app developer using Kali Linux that is mostly used for pen-testing?
Isn't it better to use Ubuntu or debian?
Ubuntu comes with an app store too, I found that very neat and useful.

**Comment by stentmartin** [comment]
I have tried your suggestion with ext4. It does not work for me!

The AParted program is not stable. It can not handle NTFS, exFAT and ext4 properly on any large SD Card. I tried a 32GB SD Card and could only format it in F32. My ext4 was made properly but disappeared when the adds come up and ended up as empty space on the SD Card.

Everything else is set to your suggested specifications.

Please advise.

stentmartin



On May 8, 2017, at 6:37 PM, Somebodyhura <notifications@github.com> wrote:

By the way, you said your developing a medical app...what type? What is it?

And I'm just too curious...why an app developer using Kali Linux that is mostly used for pen-testing?
Isn't it better to use Ubuntu or debian?
Ubuntu comes with an app store too, I found that very neat and useful.

**Comment by stentmartin** [comment]
You are right and I will use Debian in my final attempt. Right now I am focusing on something that will work and is relative stable.

On May 8, 2017, at 6:37 PM, Somebodyhura <notifications@github.com> wrote:

By the way, you said your developing a medical app...what type? What is it?

And I'm just too curious...why an app developer using Kali Linux that is mostly used for pen-testing?
Isn't it better to use Ubuntu or debian?
Ubuntu comes with an app store too, I found that very neat and useful.

**Comment by Somebodyhura** [comment]
@stentmarting

Open Aprated and then partition your SD card

Before tapping on "Apply" go to your mobile settings and eject the SD card

Now click on apply and let Aparted do its job

It may even request you to take out and put in your SD card again.

After partition is complete again take out your SD card and put it in again.

Ignore "SD card corrupt" and if your phone found the sdcard, tap on "use as portable Storage" so the phone won't format it again

Ihlgjore everything you see in Aparted after you formatted the SD card (yeah it may seem free space but it's NOT, it's ok just proceed)

Then follow as I said and it should work fine.
